src/cow-react/modules/limitOrders/containers/LimitOrdersWidget/index.tsx

```diff
@@ -60,31 +57,33 @@ export function LimitOrdersWidget() {
     (field: Field, currency: Currency) => {
       const inputCurrencyId = (field === Field.INPUT ? currency.symbol : inputCurrency?.symbol) || ''
       const outputCurrencyId = (field === Field.OUTPUT ? currency.symbol : outputCurrency?.symbol) || ''
-      const route = parameterizeLimitOrdersRoute(chainId, inputCurrencyId, outputCurrencyId)
 
-      history.push(route)
+      stateManager.navigate(chainId, inputCurrencyId, outputCurrencyId)
```

Comment:
The actual management of the history should be handled in this custom hook, instead of inside the state manager